### ISSUE TITLE
Add more rules to the static analysis tools

### ISSUE THREAD

**Issue by wkurniawan07** [TEAMMATES/teammates#5280]
With static analysis tools up and running we can now properly enforce certain coding standards. Currently the rules are not quite there yet. We shall add the rules progressively (in multiple PRs, no need to create new issue for new specific rules to be added), in the following manner:

1. One PR will concern only one tool. Adding multiple rules are fine, but try to keep the diff minimum (ideally < 1000 LoC).
2. Title of PR: `Add new rules to {toolType}`, where `{toolType}` = CheckStyle, PMD, FindBugs, ESLint. In the PR description briefly describe the rule(s) added, and don't mention `Fixes #....`.
3. Branch name: `add-new-rules-to-{tooltype}-{n}`, where `{tooltype}` = checkstyle, pmd, findbugs, eslint, and `{n}` = an arbitrary number (but try to keep it in order) signifying the progress, and also to prevent conflicts when multiple persons are working on the same toolset.
4. Merging will use the PR number instead of the issue number.

@JosephineKwa @kanghj will need your help (anyone who wants to lend a hand is also welcome!) to get this done pronto! @bamboo3250 if you're free as well :smirk:

**Comment by wkurniawan07** [comment]
We might need some mods to the workflow. In particular there is another alternative that we can stick to one rule per PR.

Pros:
1. We can standardize the branch name, e.g `add-new-{tooltype}-rule_rule-name`.
2. The PR can be kept small.

Cons:
1. When the rules added has only very few changes, it might not be so worth it (enforcing some rules will only cause < 10 lines of changes, thanks to everyone who upholds it!).
2. In fact, some rules can be added without any changes. Then the PR will be a one-liner change.
3. We'll easily reach 100-200 new PRs if we do one rule per PR.

@kanghj @JosephineKwa @damithc need your opinions on this.

**Comment by wkurniawan07** [comment]
@kanghj @JosephineKwa A tip for de-conflicting: Different tools can touch the same line in a different manner, so conflicts will be inevitable. For example, consider this line:
```
for(int i=0; i<3 ; i++ ){
```
Person A adds the rule to fix whitespace _around_ the parentheses:
```
for (int i=0; i<3 ; i++ ) {
```
Person B adds the rule to fix whitespace _inside_ the parentheses:
```
for(int i = 0; i < 3 ; i++){
```
Person A merges his change first. Person B pulls, conflict occurs.
In this case, person B will simply have to reset to Person A's work (sacrifice his own work), because Person B can simply run the static analysis once again to figure out what he needs to do (more like, what he has done before).

tl;dr: resolve conflict using "Theirs", because you know what you're going to do :smiley:

**Comment by damithc** [comment]
I'm ok with doing multiple *related* rules in the same PR if the resulting diff is not too big.

**Comment by wkurniawan07** [comment]
Will break this down to 3 separate issues for easier tracking.
